idseq_pipeline/commands/host_filtering_functions.py

```diff
@@ -598,6 +600,14 @@ def run_stage1(lazy_run=True):
         write_to_log("Number of input files was neither 1 nor 2. Aborting computation.")
         return # only support either 1 file or 2 (paired) files
 
+    # Truncate files if they are too long
+    for f in fastq_files:
+        initial_size = os.path.getsize(f)
+        execute_command("sed -i '%s,$ d' %s" % (str(MAX_INPUT_LINES + 1), f))
+        final_size = os.path.getsize(f)
+    if final_size < initial_size:
```

Comment:
this should be in the 'for' block right?

Reply:
Actually no - if there are multiple fastq_files, it should still only upload one "input_truncated" file (number of lines in all fastq_files is expected to be the same). Good point though, it would be better to make the assumption of equal line counts explicit using an assert statement.

Reply:
Yeah let's do some input validation as part of the streaming download and uncompress.   I had added that in my pending PR but reverted due to nontrivial error handling implications;   so will defer for future.

Reply:
This is how I am changing this in my pending commit, partly to address Jonathan's concern, but also changing the truncation to be to number of reads (which is 1/2 number of lines for fasta and 1/4 for fastq).   Note this number is multipled by the number of files as per our usual convention of counting a paired end read as 2 reads.  Is that useful, Charles, or should I keep it number of lines?
```
    # Truncate files if they are too long
    initial_sizes = []
    for f in fastq_files:
        initial_sizes.append(os.path.getsize(f))
        execute_command("sed -i '%s,$ d' %s" % (str(MAX_INPUT_LINES + 1), f))
        final_sizes.append(os.path.getsize(f))
    if any(fsz < isz for fsz, isz in zip(final_sizes, initial_sizes)):
        execute_command("echo %d | aws s3 cp - %s/%s" % (MAX_INPUT_READS * len(fastq_files), SAMPLE_S3_OUTPUT_PATH, INPUT_TRUNCATED_FILE))

```

Reply:
@boris-dimitrov This is nice in that it is able to handle single vs paired inputs. But your comment made me realized that fastas are not handled correctly in either your implementation or mine. The fastq format always has 4 lines per record. The fasta format allows the the sequence to be split over multiple lines, so the records could potentially have variable numbers of lines.

Reply:
Yikes.   Is it true the only fastas are from Joe's pipeline?

Reply:
Seems like in the fasta case we should truncate the file on the position of the N+1'st '>' character?   We can create an issue for that.

Reply:
I am marking it with TODO in the code.

Reply:
Yes. Thanks